dfusion_rust_core/src/models/standing_order.rs

```diff
@@ -123,31 +128,67 @@ impl From<&Arc<Log>> for StandingOrder {
 }
 
 
-impl From<Entity> for StandingOrder {
-    fn from(entity: Entity) -> Self {
-        let account_id = u16::from_entity(&entity, "accountId");
-        let batch_index = U256::from_entity(&entity, "batchIndex");
-        let valid_from_auction_id = U256::from_entity(&entity, "validFromAuctionId");
-
-        StandingOrder {
-            account_id,
-            batch_index,
-            valid_from_auction_id,            
-            orders: vec![] // Orders cannot be reconstructed just from the entity
+impl From<(Entity, Vec<Entity>)> for StandingOrder {
+    fn from(entities: (Entity, Vec<Entity>)) -> Self {
+        let batch_entity = entities.0;
+        let mut order_entities_iter = entities.1.into_iter();
+        let order_ids_value = batch_entity.get("orders").expect("it should contain order ids");
+
+        if let Value::List(order_ids) = order_ids_value {
+            assert_eq!(
+                order_ids.len(),
+                order_entities_iter.len(),
+                "The entity should have the same number of orders as Vec<Entity>"
+            );
+
+            let account_id = u16::from_entity(&batch_entity, "accountId");
```

Comment:
ubernit: since not read anywhere else, maybe inline below?

Reply:
I think this can be subjective

For me is more clear like this, this way, you separate the part where you extract the data from the construction of the object. 

I don't have a strong opinion here